x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeResponse.java

```diff
@@ -22,8 +23,31 @@
 final class ComputeResponse extends TransportResponse {
     private final List<DriverProfile> profiles;
 
+    // for use with ClusterComputeRequests (cross-cluster searches)
+    private final TimeValue took;  // overall took time for a specific cluster in a cross-cluster search
+    public final int totalShards;
+    public final int successfulShards;
+    public final int skippedShards;
+    public final int failedShards;
+
     ComputeResponse(List<DriverProfile> profiles) {
+        this(profiles, null, null, null, null, null);
+    }
+
+    ComputeResponse(
```

Comment:
I'm curious to see how this all works with INLINESTATS which runs a two phased execution. We're able to preserve the profiles for them with some work but this is outside of all that.

Reply:
As discussed in another forum, I'm not able to test this currently since INLINESTATS itself is not yet cross-cluster enabled, as indicated here: https://github.com/elastic/elasticsearch/issues/110923. We will not block this ticket on that issue and instead handle it as part of the INLINESTATS work.

Reply:
:+1:

INLINESTATS is too busted at the moment to test this with properly

Reply:
Sorry for the distraction.